### PR TITLE
Breaking up test and Making DiracFile work in test

### PR DESCRIPTION
**Pull request by rob-c**
Fixes recent minor bug in DiracFile. Better splits up the test into the components which require external resources to run.

### CODE REVIEW

**Comment by rob-c** [comment]
Any objections to merging?

File: python/GangaDirac/Lib/Files/DiracFile.py [hunk 1]
```diff
@@ -193,35 +190,38 @@ def locations(self):
...
-        if _lfn != "" and _namePattern != "":
-            self.lfn = _lfn
-            self.remoteDir = os.path.dirname(_lfn)
-            self.namePattern = os.path.basename(_namePattern)
-            self.localDir = os.path.dirname(expandfilename(_namePattern))
-
-        elif _lfn != "" and _namePattern == "":
-            self.lfn = _lfn
+        lfn = copy.deepcopy(_lfn)
+        namePattern = copy.deepcopy(_namePattern)
```
Comment (milliams) [inline python/GangaDirac/Lib/Files/DiracFile.py]: Is `_namePattern` a string? If so then it's immutable and there should be no need to do a deepcopy.

**Comment by rob-c** [comment]
again any objections?

File: python/GangaDirac/Lib/Files/DiracFile.py [hunk 2]
```diff
@@ -193,35 +190,38 @@ def locations(self):
...
-        if _lfn != "" and _namePattern != "":
-            self.lfn = _lfn
-            self.remoteDir = os.path.dirname(_lfn)
-            self.namePattern = os.path.basename(_namePattern)
-            self.localDir = os.path.dirname(expandfilename(_namePattern))
-
-        elif _lfn != "" and _namePattern == "":
-            self.lfn = _lfn
+        lfn = _lfn
+        namePattern = _namePattern
```
Comment (milliams) [inline python/GangaDirac/Lib/Files/DiracFile.py]: Why not just 
```python
def _setLFNnamePattern(self, lfn="", namePattern=""):
```
?

File: python/GangaDirac/Lib/Files/DiracFile.py [hunk 3]
```diff
@@ -193,35 +190,38 @@ def locations(self):
...
+            ## TODO REPLACE THIS WITH IN LIST OF VONAMES KNOWN
+            # Check for /lhcb/some/path or /gridpp/some/path
+            if namePattern.split(os.pathsep)[0] == self.defaultSE \
+                or (len(namePattern) > 3 and namePattern[0:4].upper() == "LFN:"\
+                    or len(namePattern.split(os.pathsep)) > 1 and namePattern.split(os.pathsep)[1] == self.defaultSE):
+                # Check for LFN:/gridpp/some/path or others...
+                lfn = namePattern
+                namePattern = ""
+
+        if lfn != "" and lfn is not None:
```
Comment (milliams) [inline python/GangaDirac/Lib/Files/DiracFile.py]: Maybe just:
```python
if lfn:
```
?

**Comment by rob-c** [comment]
Once more

**Comment by milliams** [comment]
:+1:

**Pull request merged**